adding text to missing areas (#35)
SpencerDawkins/net-collab-rqmts: draft-kwbdgrr-tsvwg-net-collab-rqmts.md

Such a mechanism might be simple, for example, a cellular network might allow one flow from a subscriber to declare itself as important; other flows with that subscriber are denied attempts to prioritize themselves. However, the network cannot identify whether the prioritized flow is legitimate or malicious.

Requirements:

>   REQ-SIGNAL-VALIDATION: The network/OS needs to ensure that the user/client signaling of priority is unique for every type of traffic within the same flow, thereby avoiding prioritizing of all the streams/traffic the same way.

Author: If the client/user signals metadata A to be highest priority and then signals metadata B to be highest priority, do we just remove A (assign it default priority) and make B higher priority or discard the whole priority table and reset all the traffic to default values?

>   REQ-SIGNAL-VALIDATION: The network/OS needs to ensure that the user/client signaling of priority is unique for every type of traffic within the same flow, thereby avoiding prioritizing of all the streams/traffic the same way.

Reviewer: ```suggestion
  REQ-SIGNAL-VALIDATION: The network/OS needs to ensure that the user/client signaling of priority (if any) does not associate the same priority level with all traffic types within the same flow, thereby avoiding prioritizing of all the streams/traffic the same way.
```

>   REQ-CLIENT-VALIDATION: The network needs to ensure the signal is coming from the same user/client that is part of the 5-tuple flow. This is to ensure no other application influences the priority of another application's flow.

Author: I purposely didn't include any signaling or host involvement in the process so as not to get into implementation details.

Reviewer 2: line 428 / requirement is not clear how it would work. Assuming that the e2e flow is encrypted, what would the user/client be able to signal about 'type of traffic within a flow' (i.e., to the network every packet looks the same). For example, in QUIC, only the connection id and spin bit are not encrypted and neither has meaning for packet priority. It would seem that only the server can indicate if a packet (or frame) has higher priority.

Author: The E2E flow is encrypted (DTLS) but the signaling of metadata would lie outside of the DTLS payload... There should be a mechanism through which the network should be able to decipher the metadata (getting into the mechanism will be moving into implementation stuff and hence, I didn't get into it here). The metadata will obfuscated/encrypted by the server and the key will be sent to the client over the encrypted channel. The client would then share the key with the network/OS (in band or out of band). This is another use of client-to-network signaling that we had in the document but removed since it goes into implementation details. So left it as a requirement. Implementation would be ideally part of a different draft. Metadata can be filled into the CID for QUIC(https://datatracker.ietf.org/doc/draft-wing-cidfi/) or UDP options or anything else...

User/Client would provide a way to decipher the metadata alone and signal "Hey, treat the packets with metadata b10110 with highest priority".

There would be a <metadata,priority> pair for every flow and not all the metadata should be allowed to have highest priority is what being conveyed here, although how to design that is again getting into implementation part and so I strayed away from it too.

Reviewer 2: Thanks, I follow now, and agree with the proposal

Interactive Audio/Video has long been using {{?RFC3550}} which runs over UDP.  As described in Section 2.3.7.2 of {{?RFC7478}}, there is value in differentiating between voice, video and data.
Today's video streaming is exclusively over TCP but will migrate to QUIC and eventually is likely to support unreliable transport ({{?RFC9221}}, {{?I-D.ietf-moq-transport}}).  With unreliable transport of video in RTP or QUIC, it is beneficial to differentiate the important video keyframes from other video frames.

Other applications, as mentioned in {{uc}}, such as gaming and remote desktop also benefit from differentiating their packets to the network.

> Many of these flows do not originate from a content provider's network -- rather, they originate from a peer (e.g., VoIP, interactive video, peer-to-peer gaming, Remote Desktop, CDN). Thus, the flows originate from an IP address that is not known before connection establishment, so there needs to be a way for the client to authorize the network elements to receive and hopefully to honor the metadata of those packets from a remote peer.

Reviewer: ```suggestion
Many of these flows do not originate from a content provider's network -- rather, they originate from a peer (e.g., VoIP, interactive video, peer-to-peer gaming, Remote Desktop, and CDN). Thus, the flows originate from an IP address that is not known before connection establishment, so there needs to be a way for the client to authorize the network elements to receive and hopefully to honor the metadata of those packets from a remote peer.
```
